stdlib/multiprocessing/heap.pyi

```diff
@@ -0,0 +1,32 @@
+import sys
+from _typeshed import Incomplete
+from collections.abc import Callable
+from mmap import mmap
+from typing_extensions import TypeAlias
+
+_Block: TypeAlias = tuple[Arena, int, int]
+
+__all__ = ["BufferWrapper"]
+
+class Arena:
+    size: int
+    buffer: mmap
+    if sys.platform == "win32":
+        name: str
+        def __init__(self, size: int) -> None: ...
+    else:
+        fd: int
+        def __init__(self, size: int, fd: int = ...) -> None: ...
+
+if sys.platform != "win32":
+    def reduce_arena(a: Arena) -> tuple[Callable[[int, Incomplete], Arena], tuple[int, Incomplete]]: ...
+    def rebuild_arena(size: int, dupfd: Incomplete) -> Arena: ...
```

Comment:
For the `dupfd` argument of `rebuild_arena`: we know it has to have a `detach()` method, and we know the `detach()` method needs to return an `int` (as the returned value of the `detach()` method is passed as the second argument to the `Arena` constructor). So you can do something like this:

```suggestion
    class _SupportsDetach(Protocol):
        def detach(self) -> int: ...

    def reduce_arena(a: Arena) -> tuple[Callable[[int, _SupportsDetach], Arena], tuple[int, Incomplete]]: ...
    def rebuild_arena(size: int, dupfd: _SupportsDetach) -> Arena: ...
```